doc/manuals/devel/cookbook.md

```diff
@@ -384,21 +384,25 @@ Note that `t1` is used and not `T1`. This is because Orion converts tenants to a
 ## Debug a Functional Test Case
 
 Sometimes you have the need of debugging a test case (i.e., a .test file). Maybe is an existing .test
-that has started to fail due to some modification in the code. Or maybe is a bug that has been reported
+that has started to fail due to some modification in the code. Or maybe it is a bug that has been reported
 as a new .test (the preferred way to report new bugs! :)
 
-Whatever the case, it may be useful to run Orion under a debugger (e.g. [`gdb`](https://www.gnu.org/software/gdb) or any of its [graphical front-ends](https://sourceware.org/gdb/wiki/GDB%20Front%20Ends))
-with that test as "input". In this way you can set breakpoints, use step-by-step execution, inspects 
-function call stack and variables, etc. as the program logic is executed.
+Whatever the case, it may be useful to run Orion in a debugger (e.g. [`gdb`](https://www.gnu.org/software/gdb) or any of its [graphical front-ends](https://sourceware.org/gdb/wiki/GDB%20Front%20Ends)),
+with that test as "input". That way you can set breakpoints, use step-by-step execution, inspect
+function call stack and variables etc., as the program logic is executed.
 
-The procedure is easy. Let's assume your test is `test_to_debug.test`. First, edit that file in order 
-to comment out any `brokerStart` and `brokerStop` line. In some cases it is also a good idea to
-comment out `dbDrop` lines, so you can have a look to the DB after .test ends.
+For this end, the functional test suite has a CLI that tells the suite to *not* start the contextBroker itself, but assume
+that the contextBroker is started already, and just run the tests against that "external contextBroker".
+This CLI option is called `--xbroker`.
+It is only supported for a **single test case**, i.e., you have to run `testHarness.sh` with a test file as parameter.
+Before running `testHarness.sh` you must start your "external contextBroker", running under gdb, valgrind or whatever program you prefer
+for debugging.
 
```

Comment:
I would also remark this is an "all or nothing" switch for all contextBroker processes involved in the test. I'd group that remark with the existing ones. In sum, I'd suggest to use a bullets list like this one:

```
Take into account a couple of remarks regarding `--xbroker` switch:
* It is only supported for a **single test case**, i.e., you have to run `testHarness.sh` with a test file as parameter.
* It doesn't start any contextBroker involved in the tests. In some case, you could need to debug only in one of them, letting the others to be started by the functional test suite. In this case, `--xbroker` shouldn't be just edit the .test file in order to comment out the `brokerStart` (and corresopnding`brokerStop` lines) you need to disable.
```

Reply:
In addition I think this fragment is useful: "Next, start contextBroker process under your debugger. The exact procedure depends on the particular debugger or debugger front-end being used, so we are not going into the details here". I woudn't remove it.

Reply:
It's kind of obvious ...

This phrase covers it I'd say (without treating our readers as idiots :)):

Before running `testHarness.sh` you must start your "external contextBroker", running under gdb, valgrind or whatever program you prefer
for debugging.

Reply:
I don't mind to be obvious and repeat things in a cookbook recepit :)

Having said that, I don't mind to leave that part as is. It can be adjusted in a latter PR.

Reply:
NTC

Reply:
I understand this is NTC with regards the last comment, but this one seems to be pending https://github.com/telefonicaid/fiware-orion/pull/3459#discussion_r267419326

Reply:
That part is there too.

Reply:
* It is only supported for a **single test case**, i.e., you have to run `testHarness.sh` with a test file as parameter.
* It doesn't start **any** of the contextBrokers involved in the tests (main context broker and brokers serving as context providers). In some cases, you may need to debug only one of the brokers, letting the other brokers be started by the functional test suite. In such a case, `--xbroker` should **not** be used - instead edit the .test file and comment out the `brokerStart` (and corresopnding`brokerStop` lines) you need to disable (amd remember to revert the changes afterwards!).

Reply:
I see, but not structured as a bullet list.

Reply:
The spell-checker of github showed me two typos (corresopnding + amd). commit pending for that

Reply:
Fixed in 90099a755

Reply:
In 90099a7 you see the bullet list.
Are we talking about different things?